src/common/colorspaces.c

```diff
@@ -810,19 +810,17 @@ static cmsHPROFILE dt_colorspaces_create_linear_infrared_profile(void)
 const dt_colorspaces_color_profile_t *dt_colorspaces_get_work_profile(const int imgid)
 {
   // find the colorin module -- the pointer stays valid until darktable shuts down
-  static dt_iop_module_so_t *colorin = NULL;
+  static const dt_iop_module_so_t *colorin = NULL;
   if(colorin == NULL)
   {
-    GList *modules = g_list_first(darktable.iop);
-    while(modules)
+    for(const GList *modules = g_list_first(darktable.iop); modules; modules = g_list_next(modules))
```

Comment:
darktable.iop should point to the start of the list (there seem to be places that count on that) so the g_list_first is unnecessary (and possibly gives the dangerous impression that it should be OK to save a random iop somewhere in the middle there). I know you want to be careful not to break anything inadvertently, but I actually believe a cleanup is the right time to expose incorrect assumptions in the code (and how else are you going to exposure them if not by having them trip up things)...

Reply:
Removing g_list_first is on my to-do list.  There are about 300 instances in the code base, of which about a dozen are working on returns from gtk_* functions (likely to be unnecessary, but I'll need to check the docs) and nearly all of the remainder appear to be applied to lists that dt maintains itself and thus already pointing at the start.

Reply:
Not a high priority, as you know better than I do; the performance impact of this should be almost zero. Just a matter of code compactness and readability. Which is why I mentioned it here; any space saved in the crowded `for` line is a win!